Selenium hello world

fmastalerz opened the pull request:
waiting for review

fmastalerz committed:
added dependencies to pom.xml

fmastalerz committed:
clean up dependencies; add HelloSelenium with Selenium Web Driver test

fmastalerz committed:
Selenium test re-written and change via Page Object pattern

fmastalerz committed:
added AdminPage, test if admin is logged passes

fmastalerz committed:
.gitignore re-writed

fmastalerz committed:
git cache cleared

fmastalerz committed:
added config file and ConfigLoader to project; in LoginTestPage browser driver, users login and password are taken from config

fmastalerz committed:
refactoring

fmastalerz committed:
changes in .gitignore

fmastalerz committed:
.gitignore fix

fmastalerz committed:
refactor

pkazlenka commented on src/test/java/LoginPageTest.java:
```diff
@@ -0,0 +1,43 @@
+        loginPage.password().sendKeys(userConfigLoader.getUserPassword());
+        loginPage.button().click();
+
+        assertEquals(userConfigLoader.getUserFullName(), loginPage.spanWithUsersRole().getText(), "Span should contain word: " + userConfigLoader.getUserFullName());
```
This line is too long. Split long lines into several shorter to make reviewer's life easier.

pkazlenka replied:
It's better to use String.format() instead of concatenations.

pkazlenka commented on src/test/java/config_loaders/EnvironmentConfigLoader.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public WebDriver getDriver() {
+        if (properties.getProperty("browser").equals("chrome")){
```
Use switch/case construction instead of nested IFs here.

pkazlenka commented on src/test/java/config_loaders/EnvironmentConfigLoader.java:
```diff
@@ -0,0 +1,43 @@
+            return new FirefoxDriver();
+        }
+        else {
+            //what if I got no browser in properties? how should I deal with it?
```
It's up to you. You can consider properties file as invalid and throw RunTimeError or use some default value, e.g. 'firefox'.

pkazlenka commented on src/test/java/config_loaders/UserConfigLoader.java:
```diff
@@ -0,0 +1,34 @@
+public class UserConfigLoader {
+    private Properties properties;
+
+    public UserConfigLoader() {
```
Don't repeat yourself. Constructor of this class and of EnvironmentConfigLoader differs only by properties file name. All duplicating code must be moved into separate method.

pkazlenka commented on src/test/java/pages/LoginPage.java:
```diff
@@ -0,0 +1,29 @@
+        return driver.findElement(By.id("page_x002e_components_x002e_slingshot-login_x0023_default-submit-button"));
+    }
+
+    public WebElement spanWithUsersRole() {
```
I'm pretty sure that this element is on main page, not login page.

pkazlenka commented on src/test/java/LoginPageTest.java:
```diff
@@ -0,0 +1,43 @@
+
+    @Test
+    void checkIfLoginDefaultUser() {
+        loginPage.userName().sendKeys(userConfigLoader.getUserLogin());
```
These actions/checks must be hidden under PageObject. So your LoginPage should have methods 'setLogin()', 'setPass()', 'loginWithCredentials()', etc.

pkazlenka replied:
https://martinfowler.com/bliki/PageObject.html

pkazlenka reviewed (changes requested):
Good for the first try. Please, fix the commented parts.

fmastalerz committed:
- shortened almost all long lines in code
- change concatenation to String.format()
- remove multiple if/else statement - now default driver is returned
- added HomePage to hold information about span with username
- page class logic is now in page class objects - not in test class
- added PropertiesLoader to load specific properties to loader classes - will rework this and try to apply factory method to return concrete object
